Sources/KIF/Additions/UIView-Debugging.m

```diff
@@ -8,254 +8,237 @@
             break;
             
         case UIDatePickerModeDateAndTime:
-            printf(" UIDatePickerModeDateAndTime");
+            [result appendString:@" UIDatePickerModeDateAndTime"];
             break;
             
         case UIDatePickerModeCountDownTimer:
-            printf(" UIDatePickerModeCountDownTimer");
+            [result appendString:@" UIDatePickerModeCountDownTimer"];
             break;
 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= 170400 //__IPHONE_17_4
         case UIDatePickerModeYearAndMonth:
-            printf(" UIDatePickerModeYearAndMonth");
+            [result appendString:@" UIDatePickerModeYearAndMonth"];
             break;
 #endif
     }
-    printf(")");
-    printf(" (minute interval: %s)", @(datePicker.minuteInterval).stringValue.UTF8String);
+    [result appendString:@")"];
+    [result appendFormat:@" (minute interval: %@)",  @(datePicker.minuteInterval)];
 }
 
-- (void)printAccessibilityElementsWithIndentation:(int)indent {
+- (void)_KIF_appendAccessibilityElementsWithIndentation:(int)indent result:(NSMutableString *)result {
     NSInteger numOfAccElements = self.accessibilityElementCount;
     if(numOfAccElements != NSNotFound) {
         for (NSInteger i = 0; i < numOfAccElements; ++i) {
-            [self printIndentation:indent];
+            [self _KIF_appendIndentation:indent result:result];
             UIAccessibilityElement *e = [(UIAccessibilityElement*)self accessibilityElementAtIndex:i];
-            printf("%s, label: %s", NSStringFromClass([e class]).UTF8String, e.accessibilityLabel.UTF8String);
+            [result appendFormat:@"%@, label: %@", NSStringFromClass([e class]), e.accessibilityLabel];
             if(e.accessibilityValue && e.accessibilityValue.length > 0) {
-                printf(", value: %s", e.accessibilityValue.UTF8String);
+                [result appendFormat:@", value: %@", e.accessibilityValue];
             }
             if(e.accessibilityHint && e.accessibilityHint.length > 0) {
-                printf(", hint: %s", e.accessibilityHint.UTF8String);
+                [result appendFormat:@", hint: %@", e.accessibilityHint];
             }
-            printf(", ");
-            [self printAccessibilityTraits:e.accessibilityTraits];
-            printf("\n");
+            [result appendString:@", "];
+            [self _KIF_appendAccessibilityTraits:e.accessibilityTraits result:result];
+            [result appendString:@"\n"];
         }
     }
 }
 
-- (void)printAccessibilityTraits:(UIAccessibilityTraits)traits {
-    
-    printf("traits: ");
-    bool didPrintOne = false;
+- (void)_KIF_appendAccessibilityTraits:(UIAccessibilityTraits)traits result:(NSMutableString *)result {
+
+    [result appendString:@"traits: "];
+    NSMutableArray<NSString *> *components = [NSMutableArray new];
     if(traits == UIAccessibilityTraitNone) {
-        printf("none");
-        didPrintOne = true;
+        [components addObject:@"none"];
```

Comment:
while here I simplified the logic a bit to get rid of the `didPrintOne = true;` stuff